lib/index.js

```diff
@@ -22,11 +22,24 @@ var AppsFlyer = module.exports = integration('AppsFlyer')
  */
 
 AppsFlyer.ensure(function(msg, settings) {
-  if (contains(msg.library(), 'ios') && !settings.appleAppID) {
+
+  if (!msg.device().type) {
+    return this.reject('Events to AppsFlyer must specify a device type as a contextual property');
+  }
+
+  if (msg.device().type.toLowerCase() !== 'ios' && msg.device().type !== 'android') {
```

Comment:
I'd be careful chaining like this -- prone to uncaughts if customers don't send exactly what you are looking for. Since this is server side, and they can send anything, you want to hedge against that.